nbclient/cli.py

```diff
@@ -136,16 +177,27 @@ def run_notebook(self, notebook_path: str) -> None:
         # Log it
         self.log.info(f"Executing {notebook_path}")
 
-        name = notebook_path.replace(".ipynb", "")
+        input_path = Path(notebook_path).with_suffix(".ipynb")
```

Comment:
@wpk-nist-gov Pardon my potential ignorance, but doesn't this change the input behavior?
The code was initially stripping the suffix then adding it, but this change adds the suffix indiscriminately. Seems if I pass a path to a notebook, the ipynb suffix will be duplicated.

Reply:
`with_suffix` respects if the path already has the extension.  For example:

```pycon
>>> from pathlib import Path
# with suffix already there, don't replace
>>> Path("a/b/c.ipynb").with_suffix(".ipynb")
PosixPath('a/b/c.ipynb')
# without suffix, add it
>>> Path("a/b/c").with_suffix(".ipynb")
PosixPath('a/b/c.ipynb')
```

The previous behavior was to replace, then add the suffix:

```python
name = notebook_path.replace(".ipynb", "")

input_path = f"{name}.ipynb"
```

These should be mostly equivalent...